py-rse/automate.Rmd

```diff
@@ -489,33 +423,35 @@ And yes,
 something like `$TARGET` would have been easier to understand,
 but we're stuck with it now.
 
-The next step is to replace the explicit list of prerequisites in the action
-with the automatic variable `$^`:
+The next step is to replace the explicit list of prerequisites in the recipe
+with the automatic variable `$^`,
+which expands to all prerequisites in the rule:
 
 ```make
 # Regenerate results for "Jane Eyre"
 results/moby_dick.csv : data/moby_dick.txt $(COUNT)
-        $(RUN_COUNT) $^ > $@
+	$(RUN_COUNT) $^ > $@
 ```
 
 However,
 this doesn't work.
 The rule's prerequisites are the novel and the word-counting program.
-When Make expands the action,
-the resulting command tries to process the program as if it were a data file:
+When Make expands the recipe,
+the resulting command tries to process the program `bin/countwords.py`
+as if it was a data file:
 
 ```shell
 python bin/countwords.py data/moby_dick.txt bin/countwords.py > results/moby_dick.csv
 ```
 
 Make solves this problem with another automatic variable `$<`,
-which mean "the first prerequisite".
+which mean "only the first prerequisite".
 Using it lets us rewrite our rule as:
 
 ```make
 # Regenerate results for "Jane Eyre"
 results/moby_dick.csv : data/moby_dick.txt $(COUNT)
-        $(RUN_COUNT) $< > $@
+	$(RUN_COUNT) $< > $@
 ```
 
 `$< > $@` is hard to read, even with practice.
```

Comment:
Two lines below we say "So please don't create anything this cryptic yourself". I think we should either reformulate that or consider if this section is worth excluding altogether if we don't recommend that people use it themselves anyways.

Reply:
Agree. If we don't want them to do anything cryptic, don't teach it.

Reply:
I reworded this as I think it is useful to know.